pyproject.toml

```diff
@@ -14,6 +14,14 @@ sympy = "^1.12"
 pyparsing ="~3.1.2"
 qref = "0.1.0"
 
+# A list of all of the optional dependencies, some of which are included in the
+# below `extras`. They can be opted into by apps.
+ipytree = { version = "^0.2.2", optional = true }
+ipywidgets = { version = "^8.1.2", optional = true }
+traitlets = { version = "^5.14.3", optional = true }
+
+[tool.poetry.extras]
```

Comment:
Seems to me that we would rather use `groups` to handle this.
What is the tradeoff, why would we use one over another @dexter2206 ?

Reply:
Extras are the same as with pip, and are pip installable, you can totally do:
```
pip install bartiq[jupyter]
```
On the other hand, groups are dependencies needed for development, and can be installed only using poetry.

The official docs confirm this: https://python-poetry.org/docs/managing-dependencies/